### PR TITLE
ENT-1863: Port ProductCertificateDTO to openapi spec

### PR DESCRIPTION
**Pull request by sonalidhome**
- Added specification in yaml file
 - Translator and its test changes

### CODE REVIEW

File: api/candlepin-api-spec.yaml [hunk 1]
```diff
@@ -2501,6 +2501,17 @@ components:
         enabled:
           type: boolean
 
+    ProductCertificateDTO:
+      description: DTO representing product certificates presented to the API
+      allOf:
+        - $ref: '#/components/schemas/TimestampedEntity'
+        - type: object
+          properties:
```
Comment (wolfdale) [inline api/candlepin-api-spec.yaml:2509]: Not sure why `ProductDTO` & `Id` are not included in properties.

Comment (sonalidhome) [reply api/candlepin-api-spec.yaml:2509]: Getter and Setter of Id have ignored with `@JsonIgnore` annotation. Hence removed as it is not exposed over API.
Similarly, the Product is also not exposed over API. Maybe I can keep `writeOnly` for Product in properties to set the Product.

Comment (sonalidhome) [reply api/candlepin-api-spec.yaml:2509]: As per the discussion, ignoring `ProductDTO` in properties.

**Comment by sonalidhome** [comment]
Thanks for noticing the missing mixin entry. Added now.

**Pull request merged**